map.js

```diff
@@ -14,90 +19,149 @@ let ctx;
 let hoverLocation = "";
+
+//image objects
+let topbar;
+let overlay;
+
+/**
+ * Last position of the mouse. used for rendering mouseover stuff.
+ */
+var lastMouseLoc = new Point(0,0);
 let mousedown = false;
 
 let img_Map;
 let icons = {};
 
-//TODO: These all need to be reworked to just use the JsonData trees
-let locArr = [];
-let nirnArr = [];
-let discoveredArr = []; //I know you probably hate me for adding ANOTHER array to fix later.
+/**
+ * An object that will be displayed on the map canvas.
+ */
+function MapObject(){
+    this.minX = 0;
+    this.maxX = 0;
+    this.minY = 0;
+    this.maxY = 0;
+}
+
+MapObject.prototype.contains = function(point){
+    return (this.minX < point.x && point.x < this.maxX && this.minY < point.y && point.y < this.maxY);
+}
+MapObject.prototype.width = function(){
+    return this.maxX - this.minX;
+}
+MapObject.prototype.height = function(){
+    return this.maxY - this.minY;
+}
 
 async function initMap(){
     //load map cord data
-    
-    //TODO: remove locArr and nirnArr and just use the jsondata trees
-    loadJsonData().then(()=>{
-        runOnTree(jsondata.nirnroot, x=>{if(x.cell == "Outdoors")nirnArr.push(x)});
-        runOnTree(jsondata.location, x=>locArr.push(x));
-    });
-    
-       
-    //do we still need to do this if the map is on its own page?
-    //TODO: create window here 
-    //TODO: do hide n seek stuff
 
     viewport = document.getElementById("wrapper_Map");
-    canvas = document.getElementById("canvas_Map");
+
+    canvas = document.createElement("CANVAS");
+    canvas.id = "canvas_Map";
+    canvas.width = 3544;
+    canvas.height = 2895;
+    viewport.appendChild(canvas);
     ctx = canvas.getContext("2d");
-    
     await initImgs();
+    initTopbar();
+    initOverlay();
     initListeners();
 
     //center map on imp city
-    mapX = 1700;
-    mapY = 885;
+    mapOffset = new Point(1700,885);
+
+    drawFrame();
+    console.log("map init'd");
+}
 
-    drawMap();
+function drawFrame(){
+    drawBaseMap();
+    drawMapOverlay();
+    //TODO: don't have topbar overlay map. or move topbar or something aaa idk
+    topbar.draw(ctx);
 }
 
-function drawMap(){
+/**
+ * Draw base map image.
+ */
+function drawBaseMap(){
     //Background color behind map. //prevents map from ghosting.
     ctx.beginPath();
     ctx.fillStyle = "#FBEFD5";
     ctx.rect(0,0,viewport.clientWidth,viewport.clientHeight);
     ctx.fill();
 
     //main map image.
-    ctx.drawImage(img_Map, mapX, mapY, (img_Map.width * zoomLevel), (img_Map.height * zoomLevel), 
+    ctx.drawImage(img_Map, mapOffset.x * zoomLevel, mapOffset.y * zoomLevel, (img_Map.width * zoomLevel), (img_Map.height * zoomLevel), 
                                     0, 0, img_Map.width, img_Map.height);
+}
+
 
+/*********************************
+ * OVERLAY FUNCTIONS
+ *  this is the icons n stuff on the map canvas.
+ *********************************/
+function initOverlay(){
+    overlay = {
+        locations : [],
```

Comment:
Is this just in the process of going away? Or are we sticking with these arrays?